lib/ts/recipe/session/types.ts

```diff
@@ -322,3 +418,77 @@ export type SessionInformation = {
     accessTokenPayload: any;
     timeCreated: number;
 };
+
+export type ClaimValidationResult = { isValid: true } | { isValid: false; reason?: JSONValue };
+export type ClaimValidationError = {
+    id: string;
+    reason?: JSONValue;
+};
+
+export type SessionClaimValidator = (
+    | // We split the type like this to express that either both claim and shouldRefetch is defined or neither.
+    {
+          claim: SessionClaim<any>;
+          /**
+           * Decides if we need to refetch the claim value before checking the payload with `isValid`.
+           * E.g.: if the information in the payload is expired, or is not sufficient for this check.
+           */
+          shouldRefetch: (payload: any, userContext: any) => Promise<boolean> | boolean;
+      }
+    | {}
+) & {
+    id: string;
+    /**
+     * Decides if the claim is valid based on the payload (and not checking DB or anything else)
+     */
+    validate: (payload: any, userContext: any) => Promise<ClaimValidationResult>;
+};
```

Comment:
Is there any reason to have this type splitting? Shouldn't all claim validators have a claim attached to them? And shouldn't all of them have a `shouldRefetch`? We can make it simpler by just requiring all of these in the types.